app/models/workarea/checkout/steps/payment.decorator

```diff
@@ -0,0 +1,15 @@
+module Workarea
+  decorate Checkout::Steps::Payment, with: :cim do
+
+    private
+
+    def set_credit_card(params)
+      super
+
+      return unless payment.credit_card.present?
+      payment.credit_card.tap do |card|
+        card.update_attributes(ip_address: params[:ip])
```

Comment:
So rather than grabbing this from `params`, you can just grab this from the `order` here: `order.ip_address`. That will remove the need to decorate the controller.